src/scripts/main.js

```diff
@@ -1,3 +1,255 @@
 'use strict';
 
 // write code here
+const MIN_AGE = 18;
+const MAX_AGE = 90;
+const NOTIFICATION_TIMEOUT = 2000;
+
+const tableHeader = document.querySelector('thead');
+const tableBody = document.querySelector('tbody');
+const employeeForm = document.createElement('form');
+const submitButton = document.createElement('button');
+let lastSortedColumnIndex = null;
+let activeTableRow = null;
+
+employeeForm.classList.add('new-employee-form');
+
+const formFields = [
+  { label: 'Name', name: 'name', type: 'text' },
+  { label: 'Position', name: 'position', type: 'text' },
+  {
+    label: 'office',
+    name: 'office',
+    type: 'select',
+    options: [
+      'Tokyo',
+      'Singapore',
+      'London',
+      'New York',
+      'Edinburgh',
+      'San Francisco',
+    ],
+  },
+
+  {
+    label: 'Age',
+    name: 'age',
+    type: 'number',
+    min: 1,
+  },
+
+  {
+    label: 'Salary',
+    name: 'salary',
+    type: 'number',
+    min: 0,
+  },
+];
+
+formFields.forEach((field) => {
+  const fieldLabel = document.createElement('label');
+
+  fieldLabel.textContent = `${field.label}: `;
+
+  let input;
+
+  if (field.type === 'select') {
+    input = document.createElement('select');
+
+    field.options.forEach((optionValue) => {
+      const option = document.createElement('option');
+
+      option.value = optionValue;
+      option.textContent = optionValue;
+
+      input.appendChild(option);
+    });
+  } else {
+    input = document.createElement('input');
+
+    if (field.min !== undefined) {
+      input.setAttribute('min', field.min);
+    }
+  }
+
+  input.setAttribute('name', field.name);
+  input.setAttribute('type', field.type);
+  input.setAttribute('data-qa', field.name);
+
+  fieldLabel.appendChild(input);
+
+  employeeForm.appendChild(fieldLabel);
+});
+
+submitButton.textContent = 'Save to table';
+
+employeeForm.appendChild(submitButton);
+
+document.body.appendChild(employeeForm);
+
+employeeForm.addEventListener('submit', (e) => {
+  e.preventDefault();
+
+  const formData = Array.from(e.target.elements).reduce((data, element) => {
+    if (element.tagName === 'INPUT' || element.tagName === 'SELECT') {
+      data[element.name] = isNaN(element.value)
+        ? element.value
+        : parseInt(element.value);
+    }
+
+    return data;
+  }, {});
+
+  if (!validateFrom(formData)) {
+    return;
+  }
+
+  pushNotification('success', 'Form submited successfuly');
```

Comment:
There is a typo in the message 'Form submited successfuly'. It should be 'Form submitted successfully'. This is an error related to the task requirements as it affects the user notification message.